File: solutions/minimum-scalar-product/scala/MinScalar.scala
object Minimum {

   def main(args: Array[String]) {

	val input = io.Source.fromFile("A-small-practice-minimum.in").getLines.toList
	var numberOfVectors =  input(0).toInt

	for (i <- 0 to numberOfVectors - 1) {
		var elements = input((3 * i + 1)).toInt
		var vector1 = input((3 * i + 2)).split(" ").map(_.toLong).sorted
		var vector2 = input((3 * i + 3)).split(" ").map(_.toLong).sorted.reverse
		var result = (vector1, vector2).zipped.map(_ * _).sum.toLong
		println( "Case #" + (i+1) + ": " + result )
	}
 }
}

File: solutions/minimum-scalar-product/dart/MinScalar.dart
import 'dart:io';

void main() {

    new File('/home/francisco/dart/google/bin/A-small-practice-minimum.in').readAsLines().then((List<String> input) {
       int numberOfVectors =  int.parse(input[0]);

       for (int i = 0; i < numberOfVectors; i++) {
           int elements = int.parse(input[3 * i + 1]);

           var vector1 = createArrayFromString( input[3 *i + 2] );
           var vector2 = createArrayFromString( input[3 *i + 3] );

           int result = 0;

           for (int j = 0; j < elements; j++) {
               result +=  vector1[j] * vector2[elements - j - 1];
           }

           print("Case #" + (i+1).toString() + ": " + result.toString());
       }
  });
}

createArrayFromString(String str) {
    var strArray = str.split(" ");
    var intArray = [];
    for (int i = 0; i < strArray.length; i++) {
      intArray.add(int.parse(strArray[i]));
    }
    intArray.sort();
    return intArray;
}
